IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -443,44 +561,128 @@ - (MSIDCredentialCacheItem *)tokenWithKey:(__unused MSIDCacheKey *)key
 
 // Read one or more credentials from the keychain that match the key (see credentialItem:matchesKey).
 // If not found, return an empty list without setting an error.
-- (NSArray<MSIDCredentialCacheItem *> *)tokensWithKey:(__unused MSIDCacheKey *)key
-                                           serializer:(__unused id<MSIDCacheItemSerializing>)serializer
-                                              context:(__unused id<MSIDRequestContext>)context
-                                                error:(__unused NSError **)error
+- (NSArray<MSIDCredentialCacheItem *> *)tokensWithKey:(MSIDCacheKey *)key
+                                           serializer:(id<MSIDCacheItemSerializing>)serializer
+                                              context:(id<MSIDRequestContext>)context
+                                                error:(NSError **)error
 {
-    NSArray *tokenItems = [self itemsWithKey:key context:context error:error];
-
-    if (!tokenItems)
-    {
-        return nil;
-    }
-
-    NSMutableArray<MSIDCredentialCacheItem *> *tokenList = [NSMutableArray new];
-    for (NSDictionary *dict in tokenItems)
+    NSArray<MSIDCredentialCacheItem *> *tokenList = @[];
+    
+    if ([key isKindOfClass:([MSIDDefaultCredentialCacheKey class])])
     {
-        NSData *jsonData = dict[(id)kSecValueData];
-        if (jsonData)
+        /*
+         Sync in memory cache with persistent cache at the time of look up.
+         */
+        @synchronized (self)
```

Comment:
maybe we'll want multiple reads in parallel? this might become a bottleneck.

Reply:
I think this was left by mistake because of a previous commit. The lock at the bottom layer of storage item should be enough for this code. There is still a potential issue that can occur because of a situation where one thread is in the middle of tokensWithKey to read a credential and at the same time another thread calls removeTokensWithKey. Since we do not have locks on the top layer of MSIDMacKeychainTokenCache, it could lead to issues. But on the other hand, if we add locks on the top layer, then we do not need any locks on the MSIDMacCredentialStorageItem item layer as all calls to the storage item are done from the top layer. We can discuss this in a follow up PR. Now there will be 4 follow-ups.:)